instrumentation/net_http/test/opentelemetry/instrumentation/net/http/instrumentation_test.rb

```diff
@@ -121,6 +121,43 @@
         headers: { 'Traceparent' => "00-#{span.hex_trace_id}-#{span.hex_span_id}-01" }
       )
     end
+
+    describe 'hooks' do
+      let(:response_body) { 'abcd1234' }
+      let(:headers_attribute) { 'headers' }
+      let(:response_body_attribute) { 'response_body' }
+
+      before do
+        stub_request(:get, 'http://example.com/body').to_return(status: 200, body: response_body)
+        instrumentation.instance_variable_set(:@installed, false)
+        config = {
+          request_hook: lambda do |span, request, _request_body|
+            headers = {}
+            request.each_header do |k, v|
+              headers[k] = v
+            end
+            span.set_attribute(headers_attribute, headers.to_json)
+          end,
+          response_hook: lambda do |span, response|
+            span.set_attribute(response_body_attribute, response.body)
+          end
+        }
+
+        instrumentation.install(config)
+      end
+
+      it 'collects data in request hook' do
```

Comment:
Unless I am missing something, are there any test cases that cover the exception flow for when the request and response hooks fail?

Reply:
Added